source/faq/storage.txt

```diff
@@ -116,63 +116,64 @@ The data files in your data directory, which is the :file:`/data/db`
-  fragmentation. MongoDB names the first data file ``<databasename>.0``,
-  the next ``<databasename>.1``, etc. The first file :program:`mongod`
-  allocates is 64 megabytes, the next 128 megabytes, and so on, up to
-  2 gigabytes, at which point all subsequent files are 2
-  gigabytes. The data files include files with allocated space but
-  that hold no data. :program:`mongod` may allocate a 1 gigabyte data
-  file that may be 90% empty. For most larger databases, unused
-  allocated space is small compared to the database.
-
-- The :term:`oplog`.
-
-  If this :program:`mongod` is a member of a replica set, the data
-  directory includes the :term:`oplog.rs <oplog>` file, which is a
-  preallocated :term:`capped collection` in the ``local``
-  database. The default allocation is approximately 5% of disk space
-  on 64-bit installations, see :ref:`Oplog Sizing
-  <replica-set-oplog-sizing>` for more information. In most cases, you
-  should not need to resize the oplog. However, if you do, see
-  :doc:`/tutorial/change-oplog-size`.
-
-- The :term:`journal`.
-
-  The data directory contains the journal files, which store write
-  operations on disk prior to MongoDB applying them to databases. See
-  :doc:`/core/journaling`.
-
-- Empty records.
-
-  MongoDB maintains lists of empty records in data files when
-  deleting documents and collections. MongoDB can reuse this space,
-  but will never return this space to the operating system.
-
-  To de-fragment allocated storage, use :dbcommand:`compact`, which
-  de-fragments allocated space. By de-fragmenting storage, MongoDB
-  can effectively use the allocated space. :dbcommand:`compact`
-  requires up to 2 gigabytes of extra disk space to run. Do not
-  use :dbcommand:`compact` if you are critically low on disk space.
-
-  .. important:: :dbcommand:`compact` only removes fragmentation
-     from MongoDB data files and does not return any disk space to
-     the operating system.
-
-  To reclaim deleted space, use :dbcommand:`repairDatabase`, which
-  rebuilds the database which de-fragments the storage and may release
-  space to the operating system. :dbcommand:`repairDatabase` requires
-  up to 2 gigabytes of extra disk space to run. Do not use
-  :dbcommand:`repairDatabase` if you are critically low on disk space.
-
-  .. warning::
-     :dbcommand:`repairDatabase` requires enough free disk space to
-     hold both the old and new database files while the repair is
-     running. Be aware that :dbcommand:`repairDatabase` will block
-     all other operations and may take a long time to complete.
+Preallocated data files
+~~~~~~~~~~~~~~~~~~~~~~~
+
+MongoDB preallocates its data files, in part to avoid file system fragmentation.
+The first file that :program:`mongod` creates, ``<databasename>.1``, will be 64
+megabytes. The second, ``<databasename>.2``, will be 128 megabytes; and so on,
+up to 2 gigabytes.
+
+The proportion of the preallocated files that is unused is typically small.
```

Comment:
This paragraph, read in context of the previous paragraph, has the following logic trail:
1. MongoDB prealllocates data files.  For a database, the first data file  = 64 Megabytes.
2. The proportion of the preallocated files that is unused is typically small.
3. However, smallfiles reduce size of preallocated files.  

2 needs qualifications for it to be consistent with 1.
How is 3 a "however" of 2 ?